File: 3/5.py
def main(s):
    result = ""
    hash_counter = {}

    for i in range(len(s)):
        if not hash_counter.get(s[i]):
            hash_counter[s[i]] = 1
            continue
        hash_counter[s[i]] += 1

    for key in hash_counter:
        result += f"{key}{hash_counter[key] if hash_counter[key] > 1 else ''}"

    return result


print(main("KKHSSSSSSSE"))

File: 3/5.js
// 시간 복잡도: O(n)
// 공간 복잡도: O(n)
function main(s) {
  let result = "";
  const hashCounter = {};

  for (let i = 0; i < s.length; i++) {
    if (!hashCounter[s[i]]) {
      hashCounter[s[i]] = 1;
      continue;
    }
    hashCounter[s[i]]++;
  }

  for (const key in hashCounter) {
    result += `${key}${hashCounter[key] === 1 ? "" : hashCounter[key]}`;
  }

  return result;
}

console.log(main("KKHSSSSSSSE"));
